internal/policy/policy.go

```diff
@@ -449,3 +489,81 @@ func (s *State) GetRootMetadata() (*tuf.RootMetadata, error) {
 
 	return rootMetadata, nil
 }
+
+func (s *State) GetTargetsMetadata(roleName string) (*tuf.TargetsMetadata, error) {
+	var (
+		payloadBytes []byte
+		err          error
+	)
+
+	if roleName == TargetsRoleName {
+		payloadBytes, err = s.TargetsEnvelope.DecodeB64Payload()
+	} else {
+		if env, ok := s.DelegationEnvelopes[roleName]; ok {
+			payloadBytes, err = env.DecodeB64Payload()
+		} else {
+			return nil, ErrMetadataNotFound
+		}
+	}
+
+	if err != nil {
+		return nil, err
+	}
```

Comment:
A good best practice is to simplify conditions and avoid nesting by trying to fail fast around exit cases. i.e. this could be re-written as:

```suggestion
	e := s.TargetsEnvelope
	if roleName != TargetsRoleName {
		env, ok := s.DelegationEnvelopes[roleName]
		if !ok {
			return nil, ErrMetadataNotFound
		}
		e = env
	}
	payloadBytes, err := e.DecodeB64Payload()
	if err != nil {
		return nil, err
	}
```